specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/Registry/preview/2025-05-01-preview/containerregistry.json

```diff
@@ -6157,1426 +8089,1397 @@
+          "x-ms-secret": true
+        },
+        "customHeaders": {
+          "type": "object",
+          "description": "Custom headers that will be added to the webhook notifications.",
+          "additionalProperties": {
+            "type": "string"
           }
         },
-        "expiry": {
-          "format": "date-time",
-          "description": "The expiry datetime of the certificate.",
-          "type": "string"
+        "status": {
+          "$ref": "#/definitions/WebhookStatus",
+          "description": "The status of the webhook at the time the operation was called."
         },
-        "thumbprint": {
-          "description": "The thumbprint of the certificate.",
-          "type": "string"
+        "scope": {
+          "type": "string",
+          "description": "The scope of repositories where the event can be triggered. For example, 'foo:*' means events for all tags under repository 'foo'. 'foo:bar' means events for 'foo:bar' only. 'foo' is equivalent to 'foo:latest'. Empty means all events."
         },
-        "encodedPemCertificate": {
-          "description": "Base 64 encoded string of the public certificate1 in PEM format that will be used for authenticating the token.",
-          "type": "string"
+        "actions": {
+          "type": "array",
+          "description": "The list of actions that trigger the webhook to post notifications.",
+          "items": {
+            "$ref": "#/definitions/WebhookAction"
+          }
         }
-      }
+      },
+      "required": [
+        "serviceUri",
+        "actions"
+      ]
     },
-    "TokenPassword": {
-      "description": "The password that will be used for authenticating the token of a container registry.",
+    "WebhookPropertiesUpdateParameters": {
       "type": "object",
+      "description": "The parameters for updating the properties of a webhook.",
       "properties": {
-        "creationTime": {
-          "format": "date-time",
-          "description": "The creation datetime of the password.",
-          "type": "string"
-        },
-        "expiry": {
-          "format": "date-time",
-          "description": "The expiry datetime of the password.",
-          "type": "string"
-        },
-        "name": {
-          "description": "The password name \"password1\" or \"password2\"",
-          "enum": [
-            "password1",
-            "password2"
-          ],
+        "serviceUri": {
           "type": "string",
-          "x-ms-enum": {
-            "name": "TokenPasswordName",
-            "modelAsString": true
+          "format": "password",
+          "description": "The service URI for the webhook to post notifications.",
+          "x-ms-secret": true
+        },
+        "customHeaders": {
+          "type": "object",
+          "description": "Custom headers that will be added to the webhook notifications.",
+          "additionalProperties": {
+            "type": "string"
```

Comment:
missing [`x-ms-secret`](https://github.com/Azure/azure-rest-api-specs/pull/37169/files#diff-19be56fcb692553e7b0fd13bef90b90d53b604068761777689d3b241d00d5a08L7606) as 
it can't add `@secret` for `Record<string>` type
<img width="800" height="92" alt="image" src="https://github.com/user-attachments/assets/adca51ad-313c-41aa-b509-6e3f594fe598" />